// plugin-welcome/src/api/models/ControllersCoolroom.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersCoolroom
 */
export interface ControllersCoolroom {
    /**
     * 
     * @type {string}
     * @memberof ControllersCoolroom
     */
    coolroomname?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersCoolroom
     */
    coomroomtypeid?: number;
}

export function ControllersCoolroomFromJSON(json: any): ControllersCoolroom {
    return ControllersCoolroomFromJSONTyped(json, false);
}

export function ControllersCoolroomFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersCoolroom {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'coolroomname': !exists(json, 'coolroomname') ? undefined : json['coolroomname'],
        'coomroomtypeid': !exists(json, 'coomroomtypeid') ? undefined : json['coomroomtypeid'],
    };
}

export function ControllersCoolroomToJSON(value?: ControllersCoolroom | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'coolroomname': value.coolroomname,
        'coomroomtypeid': value.coomroomtypeid,
    };
}
