// plugin-welcome/src/api/models/EntCoolroomTypeEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntCoolroom,
    EntCoolroomFromJSON,
    EntCoolroomFromJSONTyped,
    EntCoolroomToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntCoolroomTypeEdges
 */
export interface EntCoolroomTypeEdges {
    /**
     * Coolrooms holds the value of the coolrooms edge.
     * @type {Array<EntCoolroom>}
     * @memberof EntCoolroomTypeEdges
     */
    coolrooms?: Array<EntCoolroom>;
}

export function EntCoolroomTypeEdgesFromJSON(json: any): EntCoolroomTypeEdges {
    return EntCoolroomTypeEdgesFromJSONTyped(json, false);
}

export function EntCoolroomTypeEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntCoolroomTypeEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'coolrooms': !exists(json, 'coolrooms') ? undefined : ((json['coolrooms'] as Array<any>).map(EntCoolroomFromJSON)),
    };
}

export function EntCoolroomTypeEdgesToJSON(value?: EntCoolroomTypeEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'coolrooms': value.coolrooms === undefined ? undefined : ((value.coolrooms as Array<any>).map(EntCoolroomToJSON)),
    };
}
